orangecontrib/prototypes/widgets/owfactoranalysis.py

```diff
@@ -70,38 +141,47 @@ def __init__(self): # __init__ je konstruktor (prva metoda ki se klice, ko se us
         table.horizontalHeader().hide()
         table.setShowGrid(False)
         box.layout().addWidget(table)
-
+        """
 
     @Inputs.data
     def set_data(self, dataset):
-        self.closeContext()
+        # self.closeContext()
         if dataset is None:
             self.sample = None
         else:
-            self.openContext(dataset.domain)  #Kaj je funkcija contexta
+            # self.openContext(dataset.domain)  # what is the function of context?
+            pass
 
         self.dataset = dataset
-        self.optionsBox.setDisabled(False)
-        self.commit.now() # Takoj poklici metodo commit
+        self.commit.now()
 
     def factor_analysis(self):
-        # Z izbranimi n_componentami izracunaj FA na self.dataset
-        result = FactorAnalysis(self.n_components).fit(self.dataset.X)
-        # Iz spremenljivke result (ki je instance nekega razreda) izlusci samo tabelo ki nas zanima
-        self.components = result.components_
+        # with chosen n_components and depending on the user-selected rotation, calculate the FA on self.dataset
```

Comment:
A nicer way to organize this function would be to first set the FA method and then call it (because the call is same for all).

```
        if self.setting_for_rotation == 0:
            fa = FactorAnalysis(self.n_components)
        elif self.setting_for_rotation == 1:
            fa = FactorAnalysis(self.n_components, rotation="varimax")
        elif self.setting_for_rotation == 2:
            fa = FactorAnalysis(self.n_components, rotation="quartimax")
        result = fa.fit(self.dataset.X)
```

Also, you have defined the constants for rotations, so you can use `NoRotation` instead of `0` etc.

However, all that differs is the value of the argument rotation. You can get rid of `if`-s, for instance like this:

```
rotation = [None, "varimax", "quartimax"][self.setting_for_rotation]
result = FactorAnalysis(self.n_components, rotation=rotation).fit(self.dataset.X)
```